activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/service/utils/AggregateErrorValidationStrategy.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2017-2020 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.activiti.cloud.services.modeling.service.utils;
+
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.activiti.cloud.modeling.api.ModelValidationError;
+import org.activiti.cloud.modeling.api.ModelValidator;
+import org.activiti.cloud.modeling.core.error.ModelingException;
+import org.activiti.cloud.modeling.core.error.SemanticModelValidationException;
+import org.springframework.lang.NonNull;
+
+public class AggregateErrorValidationStrategy<V extends ModelValidator> implements ValidationStrategy<V> {
+
+    private final static String ERROR_MESSAGE = "Semantic model validation errors encountered: %d schema violations found";
+
+    private final static String WARNING_MESSAGE = "Semantic model validation warnings encountered: %d warnings found";
+
+
+    protected List<SemanticModelValidationException> getSemanticModelValidationExceptions(@NonNull Collection<V> validators, @NonNull ValidationCallback<V> callback) {
+        final List<SemanticModelValidationException> semanticModelValidationExceptionList = new LinkedList<>();
+
+        validators.stream().forEach(modelValidator -> {
+            try {
+                callback.accept(modelValidator);
+            } catch(SemanticModelValidationException e) {
+                semanticModelValidationExceptionList.add(e);
+            }
```

Comment:
Thank you for pointing it out @erdemedeiros, it would require changing how all validators work. I created an issue Activiti/Activiti#4225 and an associated Jira task for keeping track of it.